File: Week-1/3-cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // Read change owed
    float dollars;
    do 
    {
        dollars = get_float("Change owed: ");
    } 
    while (dollars <= 0);
    // Convert to cents
    int cents = round(dollars * 100);
    // Add quarter coin count
    int coins = 0;
    while (cents >= 25)
    {
        cents -= 25;
        coins++;
    }
    // Add dime coin count
    while (cents >= 10)
    {
        cents -= 10;
        coins++;
    }
    // Add nickel coin count
    while (cents >= 5)
    {
        cents -= 5;
        coins++;
    }
    // Add rest of pennie coin count
    coins += cents;
    printf("%i\n", coins);
}

File: Week-6/3 - cash/cash.py
# Import get_float method
from cs50 import get_float
# Ask user input
while True:
    change = get_float("Change owed: ")
    if change > 0:
        break
# Change coins list
coins = [0.25, 0.1, 0.05, 0.01]
# Coin counter
coin = 0
# Calculate change
for c in coins:
    while change >= c:
        change = round(change - c, 2)
        coin += 1
# Print result
print(coin)